### PR TITLE
Implement getRawTransactionStatus

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement `getRawTransactionStatus` with the new rippled protocol buffers. 

### Context of Change

rippled implemented protocol buffer support natively, which uses a new and incompatible set of protocol buffers. The implementation in rippled was completed in: https://github.com/ripple/rippled/pull/3159.

`XpringClient` uses a decorator pattern where each implementation must conform to an interface. The interface required for the `getRawTransactionStatus` method requires it return a `RawTransactionStatus`. 

This PR implements the `getRawTransactionStatus` method of the decorator. 

*Status*

- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/Xpring-JS/pull/109)
- [ ] Create code that works with the new protocol buffers from the rippled team
   - [x] [Create a new `NetworkClient` with fakes](https://github.com/xpring-eng/Xpring-JS/pull/111)
   - [x] [Create a new `DefaultXpringClient` with tests](https://github.com/xpring-eng/Xpring-JS/pull/114/files)
   - [ ] Read only calls for new `DefaultXpringClient` (In progress)
   - [ ] Write calls for `DefaultXpringClient`
- [x] [Add a boolean parameter to switch between implementations](https://github.com/xpring-eng/Xpring-JS/pull/130)

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

No change. This new class is not exposed to the world or hooked up yet. 

## Test Plan

CI Continues to work.

### CODE REVIEW

File: test/default-xpring-client-test.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import chai, { assert } from 'chai'
```
Comment (keefertaylor) [inline test/default-xpring-client-test.ts]: This was dead code which I removed.

**Comment by keefertaylor** [comment]
I know this slightly lowers code coverage. The next PR will implement `getTransactionStatus` which uses the results of this method and adds full test coverage.

File: test/fakes/fake-network-client.ts [hunk 2]
```diff
@@ -98,7 +99,17 @@ export class FakeNetworkClientResponses {
    * Construct a default getTx response.
    */
   public static defaultGetTxResponse(): GetTxResponse {
-    return new GetTxResponse()
+    const transactionResult = new TransactionResult()
+    transactionResult.setResult('tesSUCCESS')
+
+    const meta = new Meta()
```
Comment (hbergren) [inline test/fakes/fake-network-client.ts]: What is `meta`?

Comment (keefertaylor) [reply test/fakes/fake-network-client.ts]: See “Response Format” here: https://xrpl.org/tx.html

Meta is defined as “ Various metadata about the transaction.”. The protobufs are just mirroring the existing JSON API, so we accepted their structure at face value.

**Pull request merged**